src/ServiceManagement/RecoveryServices/Commands.RecoveryServices/Service/TestAzureRecoveryServicesVaultUpgrade.cs

```diff
@@ -0,0 +1,110 @@
+    /// </summary>
+    [Cmdlet(VerbsDiagnostic.Test, "AzureRecoveryServicesVaultUpgrade")]
+    public class TestAzureRecoveryServicesVaultUpgrade : RecoveryServicesCmdletBase
+    {
+        #region Parameters
+
+        /// <summary>
+        /// Gets or sets vault type.
+        /// </summary>
+        [Parameter(Mandatory = true)]
+        [ValidateSet(
+            Constants.ASRVaultType,
+            Constants.BackupVault,
+            IgnoreCase = true)]
+        public string ResourceType { get; set; }
+
+        /// <summary>
+        /// Gets or sets vault name.
+        /// </summary>
+        [Parameter(Mandatory = true)]
+        [ValidateNotNullOrEmpty]
+        public string VaultName { get; set; }
+
+        /// <summary>
+        /// Gets or sets location of the vault.
+        /// </summary>
+        [Parameter(Mandatory = true)]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        /// <summary>
+        /// Gets or sets target resource group.
+        /// </summary>
+        [Parameter(Mandatory = true)]
+        [ValidateNotNullOrEmpty]
+        [Alias("TargetRG", "TargetRGName", "RG")]
+        public string TargetResourceGroupName { get; set; }
+        #endregion Parameters
+
+        /// <summary>
+        /// ProcessRecord of the command.
+        /// </summary>
+        public override void ExecuteCmdlet()
+        {
+            try
+            {
+                this.RecoveryServicesClient.TestVaultUpgradePrerequistes(
+                    this.VaultName,
+                    this.Location,
+                    this.ResourceType,
+                    this.TargetResourceGroupName,
+                    this.Profile.Context.Subscription.Id.ToString());
+
+                this.WriteResponse(Properties.Resources.CheckPrereqSucceded);
+                this.WriteObject(Environment.NewLine);
+            }
+            catch (Exception exception)
+            {
+                ExceptionDetails details =
+                    this.RecoveryServicesClient.HandleVaultUpgradeException(exception);
+                if (!string.IsNullOrEmpty(details.WarningDetails))
+                {
+                    this.WriteWarning(details.WarningDetails);
+                }
+
+                if (!string.IsNullOrEmpty(details.ErrorDetails))
+                {
+                    throw new InvalidOperationException(
+                        Environment.NewLine +
+                        string.Format(
+                            Properties.Resources.ConfirmVaultUpgradePrereqFailed,
+                            Properties.Resources.VaultUpgradeExceptionDetails,
+                            details.ErrorDetails));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Writes content to the screen.
+        /// </summary>
+        /// <param name="contents">Data to be printed on the screen.</param>
+        private void WriteResponse(string contents)
```

Comment:
First - why implement this method twice?  Second, the output stream is the return value of your cmdlet, this method should write to one of the other streams as previously mentioned.